test/framework/resources/k8s/resources/daemonset.go

```diff
@@ -94,3 +117,13 @@ func (d *defaultDaemonSetManager) CheckIfDaemonSetIsReady(namespace string, name
 	}, ctx.Done())
 
 }
+
+func (d *defaultDaemonSetManager) DeleteAndWaitTillDaemonSetDeleted(daemonSet *v1.DaemonSet, timeout time.Duration) error {
+	ctx := context.Background()
+
+	if err := d.k8sClient.Delete(ctx, daemonSet); err != nil {
```

Comment:
We don't wait to check if the DS is deleted. We can add a poll similar to create to actually wait and confirm if the DS is cleaned up..If there is no plan to add then maybe we can rename the func as DeleteDaemonSet...

Reply:
> We don't wait to check if the DS is deleted. 

Didn't realize this. Thanks for catching. Modified.